src/pathfinder/yapf/yapf_base.hpp

```diff
@@ -113,45 +113,39 @@ class CYapfBaseT {
 		m_veh = v;
 
 		Yapf().PfSetStartupNodes();
-		bool bDestFound = true;
 
 		for (;;) {
 			m_num_steps++;
-			Node *n = m_nodes.GetBestOpenNode();
-			if (n == nullptr) {
-				break;
-			}
+			Node *best_open_node = m_nodes.GetBestOpenNode();
+			if (best_open_node == nullptr) break;
 
-			/* if the best open node was worse than the best path found, we can finish */
-			if (m_pBestDestNode != nullptr && m_pBestDestNode->GetCost() < n->GetCostEstimate()) {
```

Comment:
Isn't this check somewhat important?
Lets imagine two routes, one shorter with a (long) low speed bridge ending at the destination and one slightly longer which also ends at the destination.
At some point it will go over the slow bridge and reach its destination while following the path. Now it won't consider the longer but faster route any more, as it has found a solution.
Or am I missing something important in this change?

Reply:
The PF always chooses the best node on the open list. In case of the long slow bridge you'll get a node with a pretty high cost. This means that there will likely be other nodes on the list with a lower cost, and those will be processed first. Those might be destination nodes as well. The slow bridge node will only stop the search if it's truly the best node on the list.

The check removed here was pretty important, but my code essentially does the same, just in a slightly different way. The destination nodes don't need any special treatment, they will eventually pop out of the open list just like any other node. This is how most A* implementations do this, at least the ones I've seen.